### PR TITLE
Testing/cypress

### LINKED ISSUE #2: [🧑‍💻] Testing : Launch Page User Flow
😊 As a user, I should see a title, description, and navigation into the site.
☹️ As a user, I should see a '404 page not found' message if I try to enter an incorrect URL path.

### LINKED ISSUE #3: [🧑‍💻] Testing : Main Page User Flow : Spacecraft View
😊 As a user, I should see a grid of all of the spacecrafts.
☹️ As a user, I should see an error message if the API is down.

### LINKED ISSUE #4: [🧑‍💻] Testing : Main Page User Flow : Navigation
😊 As a user, I should see a button that will take me back to the launch page.

### LINKED ISSUE #6: [🧑‍💻] Testing : Main Page User Flow : Selecting a Craft
😊 As a user, I should be able to select a spacecraft to see more details about it.
😊 As a user, I should see the URL path change to reflect the craft that I've chosen.

### LINKED ISSUE #7: [🧑‍💻] Testing : Spacecraft Details Page User Flow : Data Load
😊  As a user, I should be able to view pertinent details about the spacecraft I've chosen.
☹️ As a user, if the details aren't available, I should see a message like 'not available'.

### LINKED ISSUE #8: [🧑‍💻] Testing : Spacecraft Details Page User Flow : Navigation
😊 As a user, I should be able to navigate back to the main page to pick another spacecraft.
😊 As a user, I should see the URL path change to reflect the navigation.

### LINKED ISSUE #23: [🐞] Bug : Search
Search results are not updating on deletion of the search query.

### PR DESCRIPTION
**Pull request by elisebeall**
## [spacecase]

- [ ] - Feature PR [🧩]
- [ ] - Refactor PR [⚙️]
- [X] - BugFix PR [🐞]
- [ ] - User Interface PR [🖍]
- [ ] - Error Handling PR [⚠️]
- [X] - Testing PR [🧑‍💻]

**What (if any) features are you implementing?**

 - No features were implemented, but the live search functionality was fixed to render updated results upon deleting the search term.
 - Closes #23 
 - End-to-End Cypress testing has been added.  There will need to be an update to one of the details-spec tests once I have done some error handling with the empty values in my object.  Currently am rendering a random key that I knew was empty.
 - Closes #2 
 - Closes #3 
 - Closes #4 
 - Closes #6 
 - Closes #7 
 - Closes #8 

**What (if anything) did you refactor?**

 - No refactoring completed.

**Were there any issues that arose?**

 - Nope, I really feel like I've got a good handle on testing with Cypress now.

**Is there anything needs to be prioritized/errors to be addressed?**

 - Issue #27
 - That issue is causing all of the 'launchers' to not load.
 The following line of code is an example of a piece of data that needs a 'not available' message.
Screen Shot 2022-01-18 at 16 16 50

**Goals? Next steps?**

 - Finish cleaning up CSS.
 - Add [PropTypes](https://stackoverflow.com/questions/59672815/do-i-use-proptypes-if-using-context-api)
 - DEPLOY! #26 

** ⬇︎ Current Screenshots ⬇︎ **

Screen Shot 2022-01-18 at 02 55 05

Screen Shot 2022-01-18 at 16 15 31

### CODE REVIEW

**Comment by elisebeall** [comment]
@ivonne-hernandez Tagging you here just for funzies.  Hope you squash your bug! 🐞

File: cypress/integration/main-page-spec.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  it('should be able to search the spacecraft displayed by name', () => {
+    cy.fixture('spacecraft.json').as('spacecraft').then((spacecraft) => {
+      cy.intercept('GET', 'https://lldev.thespacedevs.com/2.2.0/spacecraft/?limit=1000&offset=0', {
+        body: spacecraft
+      })
+    })
+      .visit('localhost:3000/spacecraft/spacecraft')
+      .get('input').type('mer')
+      .get('.main-container').children().should('have.length', '10')
+  });
```
Comment (elisebeall) [inline cypress/integration/main-page-spec.js:47]: This should be updated to include the bug fix that allows re-rendering upon deletion.  Issue #5

File: src/components/Details.js [hunk 2]
```diff
@@ -36,6 +40,9 @@ const Details = () => {
                 <tr>
                   <td colspan="4">{details?.description}</td>
                 </tr>
+                <tr>
+                  <td>{details?.spacecraft_config.info_link}</td>
+                </tr>
```
Comment (elisebeall) [inline src/components/Details.js:45]: This will be removed.

File: src/components/Error.js [hunk 3]
```diff
@@ -1,11 +1,13 @@
 import '../css/Error.css';
 import { Fragment } from 'react';
+import Button from './Button';
 
 const Error = ({ error }) => {
```
Comment (elisebeall) [inline src/components/Error.js:5]: This just reminded me that I need to do a `PropType` here.  Thanks Code Review!

File: src/contexts/CraftContext.js [hunk 4]
```diff
@@ -7,9 +7,10 @@ const CraftContextProvider = (props) => {
   const [filteredSpacecraft, setFilteredSpacecraft] = useState([]);
 
   const searchCrafts = (query) => {
-    let filteringCrafts = filteredSpacecraft.filter(craft => {
+    let filteringCrafts = spacecraft.filter(craft => {
```
Comment (elisebeall) [inline src/contexts/CraftContext.js:10]: This is the bit of code that caused my searching bug.  Issue #23

File: src/contexts/CraftContext.js [hunk 5]
```diff
@@ -7,9 +7,10 @@ const CraftContextProvider = (props) => {
...
       return craft.name.toLowerCase().includes(query.toLowerCase());
     });
+    console.log('filteringCrafts', filteringCrafts)
```
Comment (elisebeall) [inline src/contexts/CraftContext.js:13]: Need to remove.

File: src/contexts/LaunchersContext.js [hunk 6]
```diff
@@ -7,15 +7,15 @@ const LaunchersContextProvider = (props) => {
   const [filteredLaunchers, setFilteredLaunchers] = useState([]);
 
   const searchLaunchers = (query) => {
-    let filteringLaunchers = filteredLaunchers.filter(craft => {
+    let filteringLaunchers = launchers.filter(craft => {
```
Comment (elisebeall) [inline src/contexts/LaunchersContext.js:10]: SMH

File: src/endpoints.js [hunk 7]
```diff
@@ -7,7 +7,7 @@ let env = 'lldev';
 // env = 'll';
 
 const endpoints = {
-  spacecraft: `https://${env}.thespacedevs.com/2.2.0/spacecraft/?limit=100&offset=0`,
+  spacecraft: `https://${env}.thespacedevs.com/2.2.0/spacecraft/?limit=1000&offset=0`,
```
Comment (elisebeall) [inline src/endpoints.js:10]: There are like five hundred and something results.

**Review by elisebeall — COMMENTED** [review]
This update also includes the feature/css branch that I pull into it.  Guess I'm saving a step here.
